src/pages/iv-opening-qty/forms/InventoryOpeningQtysForm.js

```diff
@@ -0,0 +1,225 @@
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { useInvalidate } from 'src/hooks/resource'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { useForm } from 'src/hooks/form'
+import { InventoryRepository } from 'src/repositories/InventoryRepository'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import CustomTextArea from 'src/components/Inputs/CustomTextArea'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { ControlContext } from 'src/providers/ControlContext'
+import { ResourceLookup } from 'src/components/Shared/ResourceLookup'
+import CustomNumberField from 'src/components/Inputs/CustomNumberField'
+
+const InventoryOpeningQtysForm = ({ labels, maxAccess, recordId, record }) => {
+  const { postRequest, getRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+
+  const invalidate = useInvalidate({
+    endpointId: InventoryRepository.InventoryOpeningQtys.qry
+  })
+
+  const { formik } = useForm({
+    maxAccess,
+    initialValues: {
+      recordId: recordId || null,
+      year: '',
+      siteId: '',
+      sku: '',
+      itemId: '',
+      itemName: '',
+      qty: '',
+      pieces: '',
+      avgWeight: '',
+      lotCategoryId: '',
+      trackBy: '',
+      pieces: ''
+    },
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      year: yup.string().required(),
+      siteId: yup.string().required(),
+      sku: yup.string().required(),
+      itemId: yup.string().required(),
+      qty: yup.number().required().min(0.01, 'Quantity must be greater than 0'),
+      avgWeight: yup.number().nullable().max(999999, 'Quantity must be less than 999999')
+    }),
+    onSubmit: async obj => {
+      try {
+        const year = formik.values.year
+        const itemId = formik.values.itemId
+        const siteId = formik.values.siteId
+
+        await postRequest({
+          extension: InventoryRepository.InventoryOpeningQtys.set,
+          record: JSON.stringify(obj)
+        })
+
+        if (!year && !siteId && !itemId) {
+          toast.success(platformLabels.Added)
+        } else toast.success(platformLabels.Edited)
+
+        formik.setFieldValue(
+          'recordId',
+
+          String(obj.year) + String(obj.itemId) + String(obj.siteId)
+        )
+
+        invalidate()
+      } catch (error) {}
+    }
+  })
+
+  const editMode = !!formik.values.recordId || !!recordId
+
+  useEffect(() => {
+    ;(async function () {
+      try {
+        if (record && record.year && record.itemId && record.siteId && recordId) {
+          const res = await getRequest({
+            extension: InventoryRepository.InventoryOpeningQtys.get,
+            parameters: `_fiscalYear=${record.year}&_itemId=${record.itemId}&_siteId=${record.siteId}`
+          })
+
+          formik.setValues({
```

Comment:
can u use setFieldValue

Reply:
no , here i want to fill the formik from get request , or i would do , setField value for each field